src/utils/style-formatter.js

```diff
@@ -25,57 +24,50 @@ const backgroundPosition = {
 
 const formatProperty = (path, setting) => `${path}: ${setting};`;
 
-const getColor = (inputColor, defaultColor, isExpression) => {
-  const color = isExpression ? colorUtils.resolveExpression(inputColor) : colorUtils.resolveColor(inputColor, palette);
-  return `${color === 'none' ? defaultColor : color}`;
+const getColor = ({ useColorExpression, colorExpression, color }, defaultColor) => {
+  const resolvedColor = useColorExpression
+    ? colorUtils.resolveExpression(colorExpression)
+    : colorUtils.resolveColor(color, palette);
+  return resolvedColor === 'none' ? defaultColor : resolvedColor;
 };
 
 export default {
   getStyles({ style, disabled, Theme, element }) {
     let styles = 'width: 100%;height: 100%;padding: 4px;';
-    const fontColor = style.useFontColorExpression ? style.fontColorExpression : style.fontColor;
-    const backgroundColor = style.useBackgroundColorExpression
-      ? style.backgroundColorExpression
-      : style.backgroundColor;
-
+    const primaryColor = colorUtils.getDefaultColor(Theme);
     palette = colorUtils.getPalette(Theme);
-    primaryColor = colorUtils.getDefaultColor(Theme);
-    styles += formatProperty('color', getColor(fontColor, '#ffffff', style.useFontColorExpression));
-    styles += formatProperty('font-size', !isNaN(style.fontSize) ? `${style.fontSize}px` : '12px');
-    styles += formatProperty('text-align', style.textAlign);
-    if (style.textStyle) {
-      style.textStyle.bold && (styles += formatProperty('font-weight', 'bold'));
-      style.textStyle.italic && (styles += formatProperty('font-style', 'italic'));
-      style.textStyle.underline && (styles += formatProperty('text-decoration', 'underline'));
-    }
-    styles += formatProperty(
-      'background-color',
-      getColor(backgroundColor, primaryColor, style.useBackgroundColorExpression)
-    );
+    // enable
     disabled && (styles += formatProperty('opacity', 0.4));
     !disabled && (styles += formatProperty('cursor', 'pointer'));
-
-    if (style.background && style.background.isUsed) {
+    // font
+    styles += formatProperty('font-size', !isNaN(style.font.size) ? `${style.font.size}px` : '12px');
```

Comment:
@miralemd All default values in style (e.g. `style.font.size`) are defined in `object-properties`. Can I expect them to be there outside sense? Otherwise I need to put a lot of sanity checks in here.

Reply:
If they're in object-properties then they will be there when the object is created regardless of environment.

The thing you need to watch out for are properties/values that are injected by the property-panel definition/component, those will only exist in Sense.

Reply:
Awesome. Making sure the unit tests are using `object-properties` for mocking.